packages/openapi-typescript-helpers/index.d.ts

```diff
@@ -77,9 +77,7 @@ export type ErrorResponse<T> = FilterKeys<
 // Generic TS utils
 
 /** Find first match of multiple keys */
-export type FilterKeys<Obj, Matchers> = {
-  [K in keyof Obj]: K extends Matchers ? Obj[K] : never;
-}[keyof Obj];
+export type FilterKeys<Obj, Matchers> = Obj[keyof Obj & Matchers];
```

Comment:
Ok, so I've narrowed down the previous issue with literal types breaking inference to this type - this commit doesn't fix that yet.
But it's definitely the culprit

https://github.com/drwpow/openapi-typescript/pull/1428/files#diff-b38a8955c23641d93d6884093dd49817f791154bd4dbe0e979b8d359bcf17e35R68

If you change this line to just then suddenly it works - don't know why tho yet!
```ts
OperationRequestBodyMediaContent<T>[MediaType]
```